### PR TITLE
Allow other users and chown to context uid/gid on create...

### PR DESCRIPTION
**Pull request by lwg-galuise**
Thanks for the excellent example on doing a passthrough class for a python based fuse sample.

I wanted to use your example as a basis for triggering actions based on the creation of new files.  When I originally ran the script I discovered quickly that the mounted FUSE folder was only accessible by the user (uid) running the Python/FUSE process.  To allow for other users (uids) of the system to be able to access / read / write the folder the `**{'allow_other': True}` flag was needed.  Without that flag the folder would look like what follows to the ls command:

```
$ ls -l /mnt/fuse-mounts
ls: cannot access '/mnt/fuse-mounts/test': Permission denied
total 0
d????????? ? ?    ?    ?            ? test

```

Next, after resolving the "allow_other" issue I noticed that all files that were created were being created with the uid & gid of the python process running the passthrough script (in my case uid/gid=0).  A reference to the fuse_get_context function allows for the retrieval of the gid/uid of the user actually calling the create file  operation.  Then a quick os.chown() operation sets the gid/uid to that of the calling (context) user.

### CODE REVIEW

File: passthrough.py [hunk 1]
```diff
@@ -128,7 +131,8 @@ def fsync(self, path, fdatasync, fh):
 
 
 def main(mountpoint, root):
-    FUSE(Passthrough(root), mountpoint, nothreads=True, foreground=True)
+    FUSE(Passthrough(root), mountpoint, nothreads=True, foreground=True, **{'allow_other': True})
```
Comment (skorokithakis) [inline passthrough.py]: Why is this called by unrolling a dict, rather than just specifying the parameter directly?

**Comment by skorokithakis** [comment]
Looks good other than the one comment, thanks!

File: passthrough.py (hunk 1 above)
Comment (lwg-galuise) [reply passthrough.py]: Lol.  It's call new and green with Python ;)  Sorry about that I come from more of a Java/C# background and am still a newbie to Python.

Comment (skorokithakis) [reply passthrough.py]: Haha, no problem, just add the parameter directly and we're good.

**Comment by gtristan** [comment]
Just a note here that this may in some cases be a security concern if glossed over. It's possibly noteworthy.

In my case, I will be using fuse to implement a sort of `fakeroot` fuse filesystem, which means that some things wont really respond properly (like device nodes) but are perfectly fine for the purpose of rolling out tarballs and running `mkfs filesys.img -d /path/to/mounted/root`

However another thing we care about in a build system is that in the real underlying filesystem, there are is no setuid `/usr/bin/sudo` lying around in a regular user's build directory, this would be a security hole which would allow other users on the same system to escalate their privileges to the building users privileges (who owns a setuid binary).

Under the fuse mount, the building user can see the file as suid, and wrap up tarballs from there and the like, but it is indeed a nice feature that by default this user's mount point is not shared with other users.

**Comment by skorokithakis** [comment]
Hmm, that's true, thanks, @gtristan. Maybe we should add a note to the readme, or to the function, and make `allow_other = False` by default?

**Comment by gtristan** [comment]
@skorokithakis, well, I dont think an example such as this needs to care *that* much, but enabling it by default in the example without any notes might be encouraging dangerous things :)

What might make the example code better would be to mention the possible keyword arguments in the code as a comment and let people do what they will with that, to be honest I wasn't aware of this allow_other parameter; the underlying fuse documentation is not really great.

**Comment by skorokithakis** [comment]
Yeah, agreed, it should be off by default with a comment about the caveats.

**Comment by neuhaus** [comment]
I don't think "`allow_other`" is an issue given that if you do it as non-root it will refuse to do so unless `user_allow_other `is specified in `/etc/fuse.conf`.

Error message:
`fusermount: option allow_other only allowed if 'user_allow_other' is set in /etc/fuse.conf`

**Comment by lwg-galuise** [comment]
@neuhaus @skorokithakis ,

I just realized I never updated that parameter in my pull request from unrolling a dictionary to a direct parameter.  Sorry for responding several years later.  @neuhaus your recent comment "woke me up" on this.

Thanks again for sharing this repo for the python example.

Greg

**Comment by skorokithakis** [comment]
Looks good to me, thank you both!

**Pull request merged**